src/core/net/SRP_CODER.md

```diff
@@ -0,0 +1,349 @@
+- `T` flag (Default TTL)
+  - `0`: Default TTL is elided - use `7200` (2 hours).
+  - `1`: Default TTL is encoded in the message using compact integer format.
+
+### Header Block Format
+
+- Message ID: Two bytes, big-endian.
+- Dispatch byte.
+- Zone (domain) name: Included if not elided (indicated by the `Z` flag).
+- Default TTL value: Encoded using compact integer format if not elided (indicated by the `T` flag).
+- Host labels: Excludes the zone name and ends with an empty label.
+
+#### Distinguishing from SRP Update Message
+
+An SRP Update message uses the following DNS Update header format (RFC2136 section 2.2):
+
+```
+                                  1  1  1  1  1  1
+    0  1  2  3  4  5  6  7  8  9  0  1  2  3  4  5
+  +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+  |                      ID                       |
+  +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+  |QR|   Opcode  |          Z         |   RCODE   |
+  +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+
+```
+
+The first 3 bytes can be used to differentiate a coded message from an SRP Update message. In the coded message, the message ID is encoded in the first two bytes, similar to a general DNS header. The fixed initial bit pattern in the header dispatch byte, when mapped to the DNS header, corresponds to `QR = 0`, `Opcode = 0b0101 = 5`, with bit 5 also set. An SRP Update message also employs `Opcode = 5` but with zero flags (`Z`). Setting bit 5 to `1` is used to identify a coded message.
+
+## General Dispatch Byte Markers
+
+Following the header, each subsequent block begins with a dispatch byte. The initial bits within a dispatch byte designate its type:
+
+- `00`: Add service
+- `01`: Remove service
+- `10`: Host
+- `110`: Footer
+
+Any unused/reserved bits in dispatch bytes MUST be set to zero during encoding and its value MUST be ignored during decoding.
+
+## Add Service Block
+
+### Add Service Dispatch
+
+```
+    0      1     2     3     4     5     6     7
+  +-----+-----+-----+-----+-----+-----+-----+-----+
+  |  0  |  0  | PT  | ST  | SUB | PRI | WGT | TXT |
+  +-----+-----+-----+-----+-----+-----+-----+-----+
+```
+
+- `PT` flag (PTR TTL) - `0`: Elide PTR TTL (use default TTL from header), `1`: TTL encoded.
+- `ST` flag (SRV/TXT TTL) - `0`: Elide SRV/TXT TLL (use default TTL from header), `1`: TTL encoded.
+- `SUB` flag (sub-type) - `0`: No sub-types, `1`: Sub-types labels encoded.
+- `PRI` flag (priority) - `0`: Elide priority (use zero), `1`: Priority encoded.
+- `WGT` flag (weight) - `0`: Elide weight (use zero), `1`: Weight encoded.
+- `TXT` flag (TXT data) - `0`: No TXT data (elided), `1`: TXT data encoded.
+
+### Add Service Block Format
+
+- "Add Service Dispatch" byte
+- PTR TTL (if not elided)
+- SRV/TXT TTL (if not elided)
+- Service instance label
+- Service name labels (excluding the zone name).
+- SubType labels (if not elided) - A list of labels, terminated by an empty label (single `0x00` byte).
```

Comment:
I'm understanding this to assume that each subtype has only a single label, which I think is always true, and so this list of labels enumerates the subtypes. It might be more readily understood if you called this "subtype list" and explained it as "A series of labels, one per subtype, termined by a single empty label." I was able to figure it out by reading RFC6763 to remind myself of how subtypes work, but I think describing it the way I suggest would eliminate the need to do that.

Reply:
Changed in new push. Thanks.